More archboard feedback

m-nash opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

m-nash committed:
more archboard feedback

m-nash committed:
merge from main

m-nash committed:
regen after merge

m-nash committed:
updates after merge

m-nash committed:
update api

azure-sdk commented:
API changes have been detected in `Azure.ResourceManager`. You can review API changes [here](https://apiview.dev/Assemblies/Review/2012654fc70948c5b16a7c66e0e58a93?diffOnly=True&diffRevisionId=b3fb4cce9f314e218a9f2ad792fcd059)

m-nash committed:
update core reference to core

m-nash committed:
update core reference in network

m-nash committed:
drop core from resources

christothes reviewed (approved):
core LGTM

m-nash committed:
more feedback

m-nash committed:
fix typo

m-nash committed:
make setapiversion virtual

azure-sdk commented:
API changes have been detected in `Azure.ResourceManager`. You can review API changes [here](https://apiview.dev/Assemblies/Review/2012654fc70948c5b16a7c66e0e58a93?diffOnly=True&diffRevisionId=7237045bd12140478ac0b1addebe3beb)

m-nash committed:
honor playback for zero polling interval in mgmt interceptor

m-nash committed:
pre generate for next autorest version with removes the helper property

azure-sdk commented:
API changes have been detected in `Azure.ResourceManager`. You can review API changes [here](https://apiview.dev/Assemblies/Review/2012654fc70948c5b16a7c66e0e58a93?diffOnly=True&diffRevisionId=7237045bd12140478ac0b1addebe3beb)

christothes commented on sdk/core/Azure.Core.TestFramework/src/Instrumentation/ManagementInterceptor.cs:
```diff
@@ -13,18 +13,20 @@ namespace Azure.Core.TestFramework
             bool modifiedAskToWait = false;
 
-            if (IsLro(invocation.Method.ReturnType))
+            if (_testMode == RecordedTestMode.Playback && IsLro(invocation.Method.ReturnType))
```
should we create a simple test to validate this with a simple LRO?

m-nash committed:
add test case to validate interceptor behavior for playback vs record

m-nash committed:
bump autorest version

m-nash committed:
merge from main

christothes commented on sdk/core/Azure.Core/tests/ManagementRecordedTestBaseTests.cs:
```diff
@@ -203,5 +203,27 @@ public void ValidateForwardClientCallDefault()
+            Assert.AreEqual("TestResourceProxy", testResource.GetType().Name);
+            if (mode == RecordedTestMode.Playback)
+            {
+                Assert.LessOrEqual(sw.ElapsedMilliseconds, 1000);
```
I think the Assert is ambiguous if both are xOrEqual to 1000

```suggestion
                Assert.Less(sw.ElapsedMilliseconds, 1000);
```

archerzz commented on sdk/core/Azure.Core.TestFramework/src/Instrumentation/OperationInterceptor.cs:
```diff
@@ -55,10 +55,17 @@ internal static object InvokeWaitForCompletion(object target, Type targetType, C
             // get the concrete instance of OperationPoller.ValueTask<Response<T>> WaitForCompletionAsync<T>(Operation<T>, TimeSpan?, CancellationToken)
             var poller = InjectZeroPoller();
-            var genericMethod = poller.GetType().GetMethods().Where(m => m.Name == PollerWaitForCompletionAsyncName).FirstOrDefault(m => m.GetParameters().Length == 3);
+            var genericMethod = poller.GetType().GetMethods().Where(m => m.Name == PollerWaitForCompletionAsyncName).FirstOrDefault(m => m.GetParameters().Length == 6);
```
Here we invoke `OperationPoller.WaitForCompletionAsync(Operation operation, TimeSpan? timeSpan?, CancellationToken cancellationToken)` which will expand to the 6 parameter version. `UdpateStatusAsync`, `HasCompleted`, `Value`, `GetRawResponse` all come from the `operation` which we mock.
So here it should be OK to invoke this method? I roll back the codes and run the test locally. It seems fine.

archerzz committed:
test(core): restore test mode after `ValidateWaitOverride`

The test mode must be restored, otherwise the test mode will be overriden, thus causing errors like below:
Failed ValidateWaitOverride(Playback) [125 ms]
  Error Message:
   System.InvalidOperationException : Unable to find environment variable CORE_TENANT_ID or TENANT_ID required by test.

azure-sdk commented:
API changes have been detected in `Azure.ResourceManager`. You can review API changes [here](https://apiview.dev/Assemblies/Review/2012654fc70948c5b16a7c66e0e58a93?diffOnly=True&diffRevisionId=c90f1c387d94479c97abf8aa10ecfebd)

archerzz commented on sdk/core/Azure.Core/tests/ManagementRecordedTestBaseTests.cs:
```diff
@@ -203,5 +203,37 @@ public void ValidateForwardClientCallDefault()
+                }
+                else
+                {
+                    Assert.GreaterOrEqual(sw.ElapsedMilliseconds, 1000);
```
The test failed on CI with the following error message. I think it's a noise, so we could add some tolerance.
> Error Message:
     Expected: greater than or equal to 1000
  But was:  999

```suggestion
                    Assert.That(sw.ElapsedMilliseconds, Is.GreaterThanOrEqualTo(1000).Within(50));
```

m-nash committed:
change to less than

m-nash committed:
pull in latest

m-nash committed:
add tolerance to the test case

m-nash committed:
simplify type access

m-nash commented:
/check-enforcer evaluate
